lib/srv/heartbeatv2.go

```diff
@@ -73,16 +108,15 @@ func NewSSHServerHeartbeat(cfg SSHServerHeartbeatConfig) (*HeartbeatV2, error) {
 		return nil, trace.Wrap(err)
 	}
 
+	fetcher := newMetadataFetcher()
```

Comment:
Maybe:

```suggestion
var metadataPtr atomic.Pointer[*metadata.Metadata]
go func() {
	md, err := metadata.Get(ctx)
	if err != nil {
		log.WithError(err).Warn("Failed to fetch SSH server metadata for heartbeat")
		return
	}
	metadataPtr.Store(md)
}()
```

and then, afterwards:

```go
		if md := metadataPtr.Load(); md != nil && md.CloudMetadata != nil {
				server.SetCloudMetadata(md.CloudMetadata)
		}

```

Reply:
+1 for Zac's comment instead of mine: https://github.com/gravitational/teleport/pull/30528#discussion_r1296247989.